### ISSUE TITLE
Unable to reproduce the results in the unsupervised setting

### ISSUE THREAD

**Issue by korawat-tanwisuth** [princeton-nlp/SimCSE#90]
Hi,

I am using a single Nvidia RTX 3090 with CUDA 11. I installed the dependencies using pip install -r requirements.txt. However, the accuracy I got in the unsupervised setting is about 1 percent lower than reported. Could you please help me figure out what I need to do to reproduce the results?

Here is the training command I used (directly from the run_unsup_example.sh). I added CUDA_VISIBLE_DEVICES=0 to use a single gpu.
![image](https://user-images.githubusercontent.com/27831302/133209859-83e246d7-0ab5-4cf9-bb88-381d1d66c0ff.png)

Here is the eval result during training.

![image](https://user-images.githubusercontent.com/27831302/133210019-2ac90d10-1bd3-4e5a-855a-987a993de87f.png)

Here is the evaluation command used. I made sure to put cls_before_pooler as documented.

![image](https://user-images.githubusercontent.com/27831302/133210148-47e1d8f7-43e8-454d-b95a-e616b9d33e9b.png)

When I ran the evaluation command, I also saw this output. Is this typical?

![image](https://user-images.githubusercontent.com/27831302/133210369-0b04b41d-a3d1-4354-bfca-8ce7402fce2e.png)

Here is the final result.

![image](https://user-images.githubusercontent.com/27831302/133210199-3ec1a3fe-05c2-46ec-8bd3-a225abe1354a.png)

**Comment by ScottishFold007** [comment]
It is likely that you have made some missteps

**Comment by korawat-tanwisuth** [comment]
@ScottishFold007  Could you please point it out? I provided all the commands and the environment I used.

**Comment by gaotianyu1350** [comment]
Hi @korawat-tanwisuth, thanks for your interest in our work! To control the variables, can you try to use our pre-trained model for evaluation first? If you can get the exact results, then it means maybe something goes off during your training. Note that different computing devices or CUDA versions can bring some randomness to the training results.

**Comment by chengzhipanpan** [comment]
Having the same problem. I only get an average score of 75.10 on BERT-base-uncased.

**Comment by korawat-tanwisuth** [comment]
@gaotianyu1350 Thank you for your response! I was able to reproduce the result from the hugging face checkpoint. It seems there might be some subtle differences in the environment that cause the variability. I try changing the seed and sometimes, the result will be 76. The average, however, is around 75.10 as @chengzhipanpan reported.

**Comment by Linda230** [comment]
> Having the same problem. I only get an average score of 75.10 on BERT-base-uncased.

Hi, I also encounter the same problem and obtain an average result of 75.10, how did you solve this problem finally? The device I used is the Pytorch1.11.0 + CUDA11.3. I appreciate much for your kind help.
